vyper/venom/passes/sccp/sccp.py

```diff
@@ -329,6 +338,40 @@ def _replace_constants(self, inst: IRInstruction):
                 if isinstance(lat, IRLiteral):
                     inst.operands[i] = lat
 
+    def _fix_phi_nodes(self):
+        visited: OrderedSet[IRBasicBlock] = OrderedSet()
+        for bb in self.cfg_phi_fix_needed:
+            self._fix_phi_bb_r(bb, visited)
+
+    def _fix_phi_bb_r(self, bb: IRBasicBlock, visited: OrderedSet[IRBasicBlock]):
+        if bb in visited:
+            return
+
+        for inst in bb.instructions:
+            if inst.opcode == "phi":
+                self._fix_phi_node_inst(inst)
+
+        visited.add(bb)
+
+        for next_bb in bb.cfg_out:
+            self._fix_phi_bb_r(next_bb, visited)
+
+    def _fix_phi_node_inst(self, phi_inst: IRInstruction):
+        if len(phi_inst.parent.cfg_in) != 1:
+            return
+
+        src_bb: IRBasicBlock = phi_inst.parent.cfg_in.first()
+        assert isinstance(src_bb, IRBasicBlock)
+
+        from_src_bb = filter(lambda x: x[0] == src_bb.label, phi_inst.phi_operands)
+        operands = list(map(lambda x: x[1], from_src_bb))
+
+        assert len(operands) == 1
+        assert isinstance(operands[0], IRVariable)
+        assert phi_inst.output is not None
+        phi_inst.output.value = operands[0]
+        phi_inst.parent.remove_instruction(phi_inst)
```

Comment:
i think it might be a bug to remove the instruction. i think the instruction is dead when the value has been propagated from previous blocks, but it might not be dead if one of the parent blocks has been pruned, but it is not a constant in the other block.